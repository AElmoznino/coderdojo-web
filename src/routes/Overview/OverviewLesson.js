// @flow

import React from 'react'
import styled from 'styled-components'
import ButtonLink from 'components/Button/ButtonLink'
import H4 from 'components/Typography/H4'

type OverviewLessonProps = {
  image: {
    alt: string,
    url: string,
  },
  lessonId: number,
  lessonNumber: number,
  text: string,
}

const Wrap = styled.div`
  background-color: ${({ theme }) => theme.colors.white};
  padding: ${({
    theme: {
      sizes: { medium, third },
    },
  }) => `${third} ${third} ${medium}`};
`

export const OverviewLesson = ({
  lessonId,
  lessonNumber,
  image,
  text,
}: OverviewLessonProps) => (
  <Wrap>
    <H4>Lektion {lessonNumber}</H4>
    <img alt={image.alt} src={image.url} />
    <p>{text}</p>
    <ButtonLink to={`/lesson/${lessonId}`}>Kom ig√•ng</ButtonLink>
  </Wrap>
)

export default OverviewLesson
