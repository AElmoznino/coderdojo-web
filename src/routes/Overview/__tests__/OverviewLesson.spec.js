import React from 'react'
import { shallow } from 'enzyme'
import OverviewLesson from '../OverviewLesson'

describe('components/OverviewLesson', () => {
  let component

  beforeEach(() => {
    component = shallow(
      <OverviewLesson
        image={{ alt: 'testAlt', url: 'testUrl' }}
        lessonId="nybÃ¶rjare-1"
        lessonNumber={1}
        text="testText"
      />,
    )
  })

  it('renders OverviewLesson', () => {
    expect(component).toMatchSnapshot()
  })
})
