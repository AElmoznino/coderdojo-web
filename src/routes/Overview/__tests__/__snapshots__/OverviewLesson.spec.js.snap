// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/OverviewLesson renders OverviewLesson 1`] = `
<styled.div>
  <styled.h4>
    Lektion 
    1
  </styled.h4>
  <img
    alt="testAlt"
    src="testUrl"
  />
  <p>
    testText
  </p>
  <Styled(Link)
    to="/lesson/nybörjare-1"
  >
    Kom igång
  </Styled(Link)>
</styled.div>
`;
